tests/Unit/API/Google/MerchantReportTest.php

```diff
@@ -0,0 +1,182 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\API\Google;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\MerchantReport;
+use Automattic\WooCommerce\GoogleListingsAndAds\Vendor\Google\Service\ShoppingContent\Date as GoogleDate;
+use Automattic\WooCommerce\GoogleListingsAndAds\Value\MCStatus;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\ProductHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\Tests\Framework\UnitTest;
+use Automattic\WooCommerce\GoogleListingsAndAds\Vendor\Google\Service\ShoppingContent;
+use Automattic\WooCommerce\GoogleListingsAndAds\Vendor\Google\Service\ShoppingContent\ProductView;
+use Automattic\WooCommerce\GoogleListingsAndAds\Vendor\Google\Service\ShoppingContent\ReportRow;
+use Automattic\WooCommerce\GoogleListingsAndAds\Vendor\Google\Service\ShoppingContent\SearchRequest;
+use Automattic\WooCommerce\GoogleListingsAndAds\Vendor\Google\Service\ShoppingContent\SearchResponse;
+use Automattic\WooCommerce\GoogleListingsAndAds\Vendor\Google\Service\ShoppingContent\Resource\Reports;
+use Automattic\WooCommerce\GoogleListingsAndAds\Vendor\Google\Exception as GoogleException;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\ShoppingContentDateTrait;
+use DateTime;
+use Exception;
+use PHPUnit\Framework\MockObject\MockObject;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class MerchantReportTest
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\API\Google
+ */
+class MerchantReportTest extends UnitTest {
+
+	use ShoppingContentDateTrait;
+
+	/** @var MockObject|ShoppingContent $shopping_client */
+	protected $shopping_client;
+
+	/** @var MockObject|ProductHelper $product_helper */
+	protected $product_helper;
+
+	/** @var MockObject|OptionsInterface $options */
+	protected $options;
+
+	/** @var MerchantReport $merchant_report */
+	protected $merchant_report;
+
+
+	/**
+	 * Runs before each test is executed.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+		$this->shopping_client          = $this->createMock( ShoppingContent::class );
+		$this->product_helper           = $this->createMock( ProductHelper::class );
+		$this->shopping_client->reports = $this->createMock( Reports::class );
+
+		$this->options         = $this->createMock( OptionsInterface::class );
+		$this->merchant_report = new MerchantReport( $this->shopping_client, $this->product_helper );
+		$this->merchant_report->set_options_object( $this->options );
+	}
+
+	/**
+	 * Creates a product view with the given id, status and expiration date.
+	 *
+	 * @param string        $mc_id The MC center id.
+	 * @param string        $status The status of the product.
+	 * @param DateTime|null $expiration_date The expiration date of the product.
```

Comment:
Do we need the `$expiration_date` parameter? When we set up the product view's in `test_get_product_view_report` it's not used.

Reply:
I use the `$expiration_date` to create a "GoogleDate", which is then used to simulate the expiration_date here:

https://github.com/woocommerce/google-listings-and-ads/pull/2261/files#diff-91e7d57ff211a17cd8684387ddd428407e3530dfca1487dd2ccc4fa022bec0eaR160

I could set a DateTime in the expiration_date response, but I believe this way allows us to simulate a more realistic scenario.